src/script/interpreter.cpp

```diff
@@ -1030,6 +1032,7 @@ class CTransactionSignatureSerializer {
         // Serialize the prevout
         ::Serialize(s, txTo.vin[nInput].prevout, nType, nVersion);
         // Serialize the script
+        assert(nInput != NOT_AN_INPUT);
```

Comment:
It should be impossible to trigger because in order for nInput to be UINT_MAX, nInputs (see the loop in Serialize() below) would have to be UINT_MAX+1 or greater which is impossible since it's an unsigned int. That's assuming the only thing that calls SerializeInput() is Serialize(), but they're both public so that may not be the case.

In any case I should add a comment documenting the impossible-because UINT_MAX+1 thing.